### PR TITLE
02 connect xmtp to game

### PR DESCRIPTION
**Pull request by poludnev**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,12 +10,16 @@
...
   "dependencies": {
     "@rainbow-me/rainbowkit": "^0.5.1",
+    "@walletconnect/web3-provider": "^1.8.0",
+    "@xmtp/xmtp-js": "^6.1.0",
     "ethers": "^5.0.0",
     "next": "^12.1.6",
     "react": "^18.1.0",
     "react-dom": "^18.1.0",
+    "sass": "^1.54.9",
     "tictactoe-ai": "^1.0.4",
```
Comment (vicglarson) [inline package.json:20]: Unused package

**Review by vicglarson — APPROVED** [review]
nice

File: context/XmtpContext.tsx [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+      state: Map<string, Conversation>,
+      newConvos: Conversation[] | undefined,
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    ): any => {
+      if (newConvos === undefined) {
+        return null;
+      }
+      newConvos.forEach((convo) => {
+        if (convo.peerAddress !== client?.address) {
+          if (state && !state.has(convo.peerAddress)) {
```
Comment (utgarda) [inline context/XmtpContext.tsx:54]:
```js
if(a){
if(b){...}
}
```
->
```js
if(a && b){...}
```

File: pages/chat/index.tsx [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+  const submitHandler: React.FormEventHandler<HTMLFormElement> = async (event) => {
+    event.preventDefault();
+    const isInNetwork = !!addressInputValue
+      ? await checkIfOnNetwork(addressInputValue)
+      : false;
+
+    if (!!isInNetwork && !!addressInputValue) {
+      const newConversation = await client?.conversations.newConversation(addressInputValue)!;
+      if (!newConversation) {
+        console.error('no conversation');
```
Comment (utgarda) [inline pages/chat/index.tsx:28]: moar details in logs pls

File: pages/chat/index.tsx [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+        return;
+      }
+      setConversation(newConversation);
+    }
+  };
+
+  const sendMessageHandler = () => {
+    const messageText = String(`Time stamp: ${Date.now()}`);
+    if (!conversation) {
+      console.warn('no conversation!');
```
Comment (utgarda) [inline pages/chat/index.tsx:38]: .

File: pages/tic-tac-toe/index.tsx [hunk 5]
```diff
@@ -16,50 +45,218 @@ function App() {
...
+        streamMessages();
+    }, [conversation, peerPlayer, currentPlayer, type]);
+
+    useEffect(() => {
+        console.log('gameState', gameState);
+        // console.log('squares', squares);
+    }, [gameState]);
+
+    useEffect(() => {
+        if (currentPlayer?.id === '0xDb0b11d1281da49e950f89bD0F6B47D464d25F91') {
```
Comment (utgarda) [inline pages/tic-tac-toe/index.tsx:206]: move defaults to config

**Pull request closed without merging**